File: 09. Graph/06. Floyd Warshall Algorithm/FloydWarshall.py
def FloydWarshall(n, dist):
    # Iterating through all intermediate nodes
    for k in range(n):
        for i in range(n):
            for j in range(n):
                if dist[i][k] != float('inf') and dist[k][j] != float('inf'):
                    dist[i][j] = min(dist[i][j], dist[i][k] + dist[k][j])

def main():
    n = 4  # Number of vertices
    dist = [
        [0, 3, float('inf'), 7],
        [8, 0, 2, float('inf')],
        [5, float('inf'), 0, 1],
        [2, float('inf'), float('inf'), 0]
    ]

    FloydWarshall(n, dist)

    print("Shortest distances between every pair of vertices:")
    for row in dist:
        print(" ".join("INF" if d == float('inf') else str(d) for d in row))

main()

File: 09. Graph/06. Floyd Warshall Algorithm/FloydWarshall.go
package main

import (
	"fmt"
)

const INF = int(1e9)

func floydWarshall(n int, dist [][]int) {
	// Iterate through all intermediate nodes
	for k := 0; k < n; k++ {
		for i := 0; i < n; i++ {
			for j := 0; j < n; j++ {
				if dist[i][k] != INF && dist[k][j] != INF {
					if dist[i][k]+dist[k][j] < dist[i][j] {
						dist[i][j] = dist[i][k] + dist[k][j]
					}
				}
			}
		}
	}
}

func main() {
	n := 4

	dist := [][]int{
		{0, 3, INF, 7},
		{8, 0, 2, INF},
		{5, INF, 0, 1},
		{2, INF, INF, 0},
	}

	floydWarshall(n, dist)

	fmt.Println("Shortest distances between every pair of vertices:")
	for i := 0; i < n; i++ {
		for j := 0; j < n; j++ {
			if dist[i][j] == INF {
				fmt.Print("INF ")
			} else {
				fmt.Print(dist[i][j], " ")
			}
		}
		fmt.Println()
	}
}